sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorConnection.java

```diff
@@ -481,6 +483,10 @@ public void dispose() {
             managementNodeCloseOperations.doFinally(signalType ->
                 logger.atVerbose()
                     .log("Closed management nodes.")))
+            .then(emitShutDownSignalOperation.doFinally(signalType ->
```

Comment:
emitShutDownSignal will trigger close session and sender/receiver. Here it can make sure session is closing after CBS + mgmt node closed.

Reply:
If we don't make emitShutDownSignal later than close RRchannel, the CBS channel may not send the link:DETACH frame and frame sequence may like:

![cbs-closing-frame-logic3](https://user-images.githubusercontent.com/97088315/162151409-67de04ea-dd7e-48cc-80d3-4a31182df10b.png)

Although it is expected behavior that session:END frame can override the link:DETACH. Do we still to have this code to make sure to send out the detach frame?

Reply:
If we put `emitShutDownSignal` ealier than `cbsCloseOperation`, because we add `.takeUntilOther(shutdownSignalSink.asMono())`, an "Cannot subscribe" exception will be thrown. Here we make `emitShutDownSignal` later than `cbsCloseOperation`. By this change, it will only request one extra AMQP channel (uninitialized) and then closed.